base/applications/cmdutils/taskkill/taskkill.c

```diff
@@ -440,6 +610,146 @@ static BOOL add_to_task_list(WCHAR *name)
     return TRUE;
 }
 
+#ifdef __REACTOS__
+
+static int get_argument_type(WCHAR* argument)
+{
+    int i;
+
+    if (argument[0] != L'/' && argument[0] != L'-')
+    {
+        return OP_PARAM_INVALID;
+    }
+    argument++;
+
+    for (i = 0; i < _countof(opList); i++)
+    {
+        if (!strcmpiW(opList[i], argument))
+        {
+            return i;
+        }
+    }
+    return OP_PARAM_INVALID;
+}
+
+/* FIXME
+argument T not supported
+*/
+
+static BOOL process_arguments(int argc, WCHAR* argv[])
+{
+    BOOL has_im = FALSE, has_pid = FALSE, has_help = FALSE;
+
+    if (argc > 1)
+    {
+        int i;
+        for (i = 1; i < argc; i++)
+        {
+            int argument = get_argument_type(argv[i]);
+
+            switch (argument)
+            {
+            case OP_PARAM_FORCE_TERMINATE:
+            {
+                if (force_termination == TRUE)
+                {
+                    // -f already specified
+                    taskkill_message_printfW(STRING_PARAM_TOO_MUCH, argv[i], 1);
+                    taskkill_message(STRING_USAGE);
+                    return FALSE;
+                }
+                force_termination = TRUE;
+                break;
+            }
+            case OP_PARAM_IMAGE:
+            case OP_PARAM_PID:
+            {
+                if (!argv[i + 1])
+                {
+                    taskkill_message_printfW(STRING_MISSING_PARAM, argv[i]);
+                    taskkill_message(STRING_USAGE);
+                    return FALSE;
+                }
+
+                if (argument == OP_PARAM_IMAGE)
+                    has_im = TRUE;
+                if (argument == OP_PARAM_PID)
+                    has_pid = TRUE;
+
+                if (has_im && has_pid)
+                {
+                    taskkill_message(STRING_MUTUAL_EXCLUSIVE);
+                    taskkill_message(STRING_USAGE);
+                    return FALSE;
+                }
+
+                if (get_argument_type(argv[i + 1]) != OP_PARAM_INVALID)
+                {
+                    taskkill_message_printfW(STRING_MISSING_PARAM, argv[i]);
+                    taskkill_message(STRING_USAGE);
+                    return FALSE;
+                }
+
+                if (!add_to_task_list(argv[++i])) // add next parameters to task_list
+                    return FALSE;
+
+                break;
+            }
+            case OP_PARAM_HELP:
+            {
+                if (has_help == TRUE)
```

Comment:
Or just `if (has_help)` if I follow the code style used elsewhere in the file & your code.

Reply:
Good idea, I will correct it.